va/validation-authority.go

```diff
@@ -800,13 +807,23 @@ func (va *ValidationAuthorityImpl) checkCAARecords(ctx context.Context, identifi
 	for _, caa := range caaSet.Issue {
 		if extractIssuerDomain(caa) == va.IssuerDomain {
 			va.stats.Inc("VA.CAA.Authorized", 1, 1.0)
-			return true, true, nil
+			return true, true
 		}
 	}
 
 	// The list of authorized issuers is non-empty, but we are not in it. Fail.
 	va.stats.Inc("VA.CAA.Unauthorized", 1, 1.0)
-	return true, false, nil
+	return true, false
+}
+
+func (va *ValidationAuthorityImpl) checkCAARecords(ctx context.Context, identifier core.AcmeIdentifier) (present, valid bool, err error) {
```

Comment:
This function moved lower in the file but it's not clear why. Can you move it back? Would make it easier to review the diff.